src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -281,48 +276,117 @@ export default class ExtensionRequest extends Vue {
   }
 
   @Watch('data.isPrevExtension')
+  @Watch('previousAgmDateText')
   @Watch('extensionExpiryDateText')
   onIsPrevExtensionChanged (): void {
+    /* eslint-disable brace-style */
+    const currentDate = DateUtilities.yyyyMmDdToDate(this.getCurrentDate)
```

Comment:
Since the calculations below depend on current date, which could change, please add it as a Watch for this method.